Jenkinsfile.release

```diff
@@ -0,0 +1,206 @@
+// -*- Groovy -*-
+/*
+  Require credentials:
+    - Microsoft Azure Service Principal
+    - Azure Storage Account
+
+  Maven docker image cannot be run with the root user
+  https://batmat.net/2017/01/30/do-not-run-your-tests-in-continuous-integration-with-the-root-user/
+  Docker container related to jenkins-jnlp require:
+  * uid 1000,
+  * gid 1000
+  * /home/jenkins/.jenkins
+
+  Jenkins Plugins:
+    * Azure-Credentials
+    * SSH-agent
+
+  docker image olblak/maven is just a customized docker image based on maven where maven is run as
+  user jenkins with uid 1000 and gid 1000, there is also a volume /home/jenkins/.jenkins defined in the image
+  otherwise the directory is created with root permission and Jenkins cannot add files there
+*/
+
+pipeline {
+    agent {
+      kubernetes {
+        label 'jenkins-release'
+        defaultContainer 'jnlp'
+        yaml '''
```

Comment:
Isn't it an obsolete way to define settings in the plugin?

Reply:
Not according the [documentation ](https://github.com/jenkinsci/kubernetes-plugin#declarative-pipeline) but to me it would make more sense to move container configuration to jenkins instance configuration instead of the Jenkinsfile which is what I mean when I say 'Todo: Move kubernetes agent configuration to the Jenkins configuration located here'

Reply:
`yamlFile` is nice for syntax highlighting at least.

Reply:
@jglick Sorry I missed your comment here, and yes I need credentials as I want to be able to trigger a release based  on private repository like jenkinsci-cert. I am going to close this PR as I created a new repository that contains, so I will probably implement your suggestion [here](https://github.com/jenkins-infra/release/blob/master/Jenkinsfile.d/release)